homeassistant/components/geniushub/binary_sensor.py

```diff
@@ -0,0 +1,79 @@
+"""Support for Genius Hub binary_sensor devices."""
+import logging
+from time import (localtime, strftime)
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from . import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+GH_IS_SWITCH = ['Dual Channel Receiver', 'Electric Switch', 'Smart Plug']
+
+GH_STATE_ATTRS = ['measuredTemperature']
```

Comment:
State attributes should be lowercase snakecase.

Reply:
This should prefrably be its own sensor if not already reported by the climate or water heater platforms.

Reply:
It is `measuredTemperature` instead of `measured_temperature` only because the underlying RESTful API calls it that, and I was hesitant to change the name:
```bash
root@vm-builder:~$ curl -X GET https://my.geniushub.co.uk/v1/devices -H "authorization: Bearer ${HUB_TOKEN}"
...
{'id': '4', 'type': 'Room Thermostat', 'assignedZones': [{'name': 'Kitchen'}], 'state': {'batteryLevel': 61, 'setTemperature': 4, 'measuredTemperature': 18.73}}, 
...
```

Reply:
> This should prefrably be its own sensor if not already reported by the climate or water heater platforms.

The Climate/WaterHeater temperatures are a _function of_ these temperatures, for example:
a RoomThermostat > RoomSensor > avg(TRVs); and also Climate zones can share 'stats, sensors, etc.

For these TRVs, I need `batteryLevel`, `lastComms` (not seen in the cURL output, above, and - to a lessor extent - `measuredTemperature`.  There are many other state attributes I can ignore for now...

I was thinking:
 1) Battery sensor (`batteryLevel`) with device state attribute of `lastComms` +/- `measuredTemperature`, _or_
 2) Temp sensor (`measuredTemperature`) with device state attribute of `lastComms` and `batteryLevel`

Option 1) produced a graph that didn't change per 24h, and so seemed a bit unsatisfactory.  Option 2) Seemed a better representation of the device to me.

I think you're suggesting:
 1) Battery sensor (`batteryLevel`), _and_
 2) Temp sensor (`measuredTemperature`)

If so, where does lastComms go?

Reply:
It can go on both.

As long as `_boiler.temperature` is not the a same as `measuredTemperature` we can add `measuredTemperature` as a separate sensor.